src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2618,6 +2618,101 @@ internal static int IndexOfAnyValueType<T>(ref T searchSpace, T value0, T value1
             return -1;
         }
 
+        public static void Replace<T>(Span<T> span, T oldValue, T newValue)
+        {
+            for (int i = 0; i < span.Length; ++i)
+            {
+                ref T val = ref span[i];
+
+                if (EqualityComparer<T>.Default.Equals(oldValue, val))
+                {
+                    val = newValue;
+                }
+            }
+        }
```

Comment:
This `Replace<T>` should probably be implemented in terms of `IEquatable<T>` rather than `EqualityComparer<T>.Default`, to help with perf for reference types.  It will mean special-casing null, though.